website/pages/blog/graphql-over-internet.mdx

```diff
@@ -0,0 +1,250 @@
+for swiftly auditing servers - even those that are running locally! It is always up-to-date with the
+current spec.
+
+Of course, security is very important - the website audits run exclusively in the browser, and there is
+no tracking, data sharing, or external dependencies! Meaning you can
+download the website on your computer and have a portable offline checker.
+
+## [GraphQL over WebSocket](https://github.com/graphql/graphql-over-http/blob/main/rfcs/GraphQLOverWebSocket.md)
+
+Before we talk about the current state of GraphQL WebSocket support, let's start with a brief
+overview of the history. Originally there was this library
+[`subscriptions-transport-ws`](https://github.com/apollographql/subscriptions-transport-ws) that was
+built and maintained by [Apollo](https://www.apollographql.com/). It also had an
+[accompanying protocol](https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md).
+But as time passed, they've been able to spend less time on it, probably due to increase demand in
+other areas. The repo grew in issues, PRs were never reviewed - it eventually became unmaintained
+and therefore deprecated. But this library actually kick-started and pushed the evolution, it was a
+big inspiration and a great influence for the upcoming refresh!
+
+[The WebSocket Protocol](https://www.rfc-editor.org/rfc/rfc6455) provides a persistent, full-duplex
+communication channel between the client and server, allowing data to be transmitted in real-time.
+Full-duplex is just a fancy word for bidirectional communication - meaning, both the server and the
+client can send each other messages through the connection channel whenever they want. When used in
+conjunction with GraphQL, it allows clients to subscribe and receive events from the server in
+real-time. This makes it ideal for applications that require real-time updates and push
+notifications.
+
+The
+[GraphQL over WebSocket protocol](https://github.com/graphql/graphql-over-http/blob/main/rfcs/GraphQLOverWebSocket.md)
+leverages all pros of the WebSocket and gives shine to real-time GraphQL. Please note that it is
+**not** backwards compatible with
+[Apollo's protocol behind `subscriptions-transport-ws`](https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md).
+
+⚠️ Beware that the specification is currently an RFC and is open to changes!
+
+### [Reference Implementation](https://github.com/enisdenjo/graphql-ws)
+
+A specification goes best with a reference implementation. Say hello to
+[`graphql-ws`](https://github.com/enisdenjo/graphql-ws). It is a zero-dependency server and client,
+very simple and bare-bone. But don't be mistaken, it's very versatile and extendable!
+
+One key element people often struggle with WebSockets are re-connections and `graphql-ws` is all
+about stable connections. Silent retries with custom strategies are easy to use and extend.
```

Comment:
(I tried to avoid double `with` — ie `struggle with with X`)
```suggestion
A common struggle when working with WebSockets is re-connections. `graphql-ws` is all
about stable connections. Silent retries with custom strategies are easy to use and extend.
```

Reply:
I guess it should be plural, as in:

> A common struggle when working with WebSockets _are_ re-connections.

or even better

> A common struggle when working with WebSockets is re-_connecting_.

What do you think?

Reply:
Yeah, the second one sounds best! 

But I still think that it should be singular because `a struggle` is singular (and the right-hand side of the sentence doesn't matter since the `struggle` is the subject of this sentence)

Reply:
You're right, it should be "is". Sounded kinda' off... :D

I chose the second one anyways. Thanks!